File: concepts/10_time_locale_and_runtime/02_calendar_time_zones_and_arithmetic/julia/test_02_transition_and_arithmetic_boundaries.jl
# polyglot-family: time_locale_and_runtime
# polyglot-concept: calendar_time_zones_and_arithmetic
# polyglot-related: languages/julia/standard_library/09_text_data_dates_and_math/
# polyglot-related+: test_069_dates_periods_and_calendar_arithmetic.jl
#
# 共同问题：DST transition 和月底算术怎样处理；缺失时区数据库时如何表达能力边界。
# 对照观察：Julia 标准库不提供 IANA zone 类型；月份算术可执行，DST 需显式第三方能力且本课程不安装。

using Test
using Dates

@testset "标准库只保证无时区 calendar arithmetic" begin
    @test Date(2024, 1, 31) + Month(1) == Date(2024, 2, 29)
    @test Date(2023, 1, 31) + Month(1) == Date(2023, 2, 28)
    @test Base.find_package("TimeZones") === nothing
    @test !isdefined(Dates, :ZonedDateTime)
end

File: concepts/10_time_locale_and_runtime/02_calendar_time_zones_and_arithmetic/rust/test_02_transition_and_arithmetic_boundaries.rs
// polyglot-family: time_locale_and_runtime
// polyglot-concept: calendar_time_zones_and_arithmetic
// polyglot-related: languages/rust/tests/tooling_and_runtime/
// polyglot-related+: 15_time_runtime_reflection_and_unsafe/test_115_timezone_calendar_and_locale_absence.rs
//
// 共同问题：DST gap/fold 如何影响本地时间；“一天”是 calendar unit 还是固定 24 小时。
// 对照观察：std has no timezone database or DST transition resolver；86,400 seconds is only fixed Duration, not local day.

use polyglot_rust_course::assert_compile_fails;
use std::time::Duration;

#[test]
fn comparison() {
    let fixed_day = Duration::from_secs(24 * 60 * 60);
    assert_eq!(fixed_day.as_secs(), 86_400);
    assert_compile_fails(
        "fn main(){ let _=std::time::TimeZone::load(\"America/New_York\"); }",
        &["could not find", "std::time"],
    );
}
